CMS/cms-basics

```diff
@@ -0,0 +1,346 @@
+          type: ImageAsset
+          label: Avatar
+        - type: _created_at
+
+```
+
+Save your CMS configurations and open your CMS again. You should be able to access the detail page of your records 🤘
+
+
+#### Edit & new
+Since the configuration of setting **editable fields** and **fill-in-able fields** when creating a new records are very similar to setting the displayable fields, we are not going into details of the two. You may simply follow the below examples.
+
+Please note that I have taken away the field _created_at because usually you may not want your users to edit the record creation time.
+
+
+```
+site:
+  - type: Record
+    name: user
+    label: User
+  - type: Record
+    name: blogpost
+    label: Blogpost
+
+records:
+  user:
+    list:
+      fields:
+        - name: username
+          type: String
+          label: Username
+        - name: role
+          type: Dropdown
+          label: Role
+        - type: _created_at
+      filters:
+        - name: role
+          type: String
+          label: Role
+     show:
+       fields:
+        - name: username
+          type: String
+          label: Username
+        - name: role
+          type: Dropdown
+          label: Role
+        - type: _created_at
+     edit:
+        fields:
+        - name: username
+          type: String
+          label: Username
+        - name: role
+          type: Dropdown
+          label: Role
+        - name: avatar
+          type: ImageAsset
+          label: Avatar
+     new:
+        fields:
+        - name: username
+          type: String
+          label: Username
+        - name: role
+          type: Dropdown
+          label: Role
+        - name: avatar
+          type: ImageAsset
+          label: Avatar
+
+```
+
+## Explaining site types in CMS
+
+Site types defines the nature of the page. For example, if you are going to show a record on the page, the type is Record.
+
+Currently there are only two types: **Record** and **PushNotifications**.
+
+PushNotification provides you with a management interface to broadcast push messages to your users. This features is coming in June.
+
+## Explaining record types in the CMS
+
+In our above examples, whenever you need to configure a field, you have to specify **its name**, **its label** and **its type**. Since **record types in the CMS is not equivalent to data type at database level**, in the section we will go into more details about the record types in CMS.
+
+Record types in the CMS refers to the type of data a user would enter. For example, while role is **String** at database level, however, as there are only two roles availabe and you want to limit the user input, you may use **Dropdown** instead.
+
+```
+records:
+  user:
+    list:
+      fields:
+        - name: role
+          type: Dropdown
+          label: Role
+        - type: _created_at
+```
+
+Here's a list of available types:
+
+* Skygear reserved columns (see **_created_at** in the above example)
```

Comment:
You may notice that only _created_at is type: _created_at, but others are name: username, because _created_at is a reserved column, while username is not.

There are three reserved column _id, _created_at and _updated_at. Not sure if it worths a note.